internal/goofys.go

```diff
@@ -1012,11 +1026,14 @@ func (fs *Goofys) MkDir(
 		return err
 	}
 
+	parent.mu.Lock()
+
 	fs.mu.Lock()
 	defer fs.mu.Unlock()
-
 	fs.insertInode(parent, inode)
```

Comment:
insertInode() has the following comments but we were not acquiring the parent lock before we call.

// LOCKS_REQUIRED(fs.mu)
// LOCKS_REQUIRED(parent.mu)